tests/test_flow/test_copy.py

```diff
@@ -0,0 +1,138 @@
+import pytest
+import pickle
+from pathlib2 import Path
+from subprocess import check_call
+from ..helpers import (
+    skip_unless_gcs, GCS_TEST_BUCKET, df_from_csv_str, equal_frame_and_index_content, dask_py3_only
+)
+
+import bionic as bn
+import dask.dataframe as dd
+
+
+@pytest.fixture(scope='function')
+def preset_builder(builder):
+    builder.assign('x', 2)
+    builder.assign('y', 3)
+
+    @builder
+    def f(x, y):
+        return x + y
+
+    return builder
+
+
+@pytest.fixture(scope='function')
+def flow(preset_builder):
+    return preset_builder.build()
+
+
+@pytest.fixture(scope='function')
+def expected_dask_df():
+    df_value = df_from_csv_str('''
+    color,number
+    red,1
+    blue,2
+    green,3
+    ''')
+    return dd.from_pandas(df_value, npartitions=1)
+
+
+@pytest.fixture(scope='function')
+def dask_flow(builder, expected_dask_df):
+
+    @builder
+    @bn.protocol.dask
+    def dask_df():
+        return expected_dask_df
+
+    return builder.build()
+
+
+def test_copy_filename(flow):
+    assert pickle.loads(flow.get('f', mode='FileCopier')
+                        .src_file_path
+                        .read_bytes()) == 5
+
+
+def test_copy_file_to_existing_local_dir(flow, tmp_path):
+    dir_path = tmp_path / 'output'
+    dir_path.mkdir()
+    flow.get('f', mode='FileCopier').copy(destination=dir_path)
+
+    expected_file_path = dir_path / 'f.pkl'
+    assert pickle.loads(expected_file_path.read_bytes()) == 5
+
+
+def test_copy_file_to_local_file(flow, tmp_path):
+    file_path = tmp_path / 'data.pkl'
+    flow.get('f', mode='FileCopier').copy(destination=file_path)
+
+    assert pickle.loads(file_path.read_bytes()) == 5
+
+
+def test_copy_file_to_local_file_using_str(flow, tmp_path):
+    file_path = tmp_path / 'data.pkl'
+    file_path_str = str(file_path)
+    flow.get('f', mode='FileCopier').copy(destination=file_path_str)
+    assert pickle.loads(file_path.read_bytes()) == 5
+
+
+@skip_unless_gcs
+def test_copy_file_to_gcs_dir(flow, tmp_path):
+    flow.get('f', mode='FileCopier').copy(destination='gs://' + GCS_TEST_BUCKET)
```

Comment:
Hmm, I realize this code was already here, so this probably shouldn't block this PR, but it'd probably be better to have separate setup and teardown steps like in `test_persistence_gcs` (see `tmp_object_path`).  Maybe we could add a TODO?